### PR TITLE
PowerModeButton: rewrite as radiobuttons

### PR DESCRIPTION
**Pull request by danrabbit**
## Before
Screenshot from 2022-01-10 11 33 21

## After
Screenshot from 2022-01-10 11 32 56

### CODE REVIEW

File: src/Widgets/PowerModeButton.vala [hunk 1]
```diff
@@ -19,72 +19,104 @@
...
+            switch (pprofile.profiles[i].get ("Profile").get_string ()) {
+                case "power-saver":
+                    add (saver_radio);
+                    break;
+                case "balanced":
+                    add (balanced_radio);
+                    break;
+                case "performance":
+                    add (performance_radio);
+                    break;
```
Comment (jeremypw) [inline src/Widgets/PowerModeButton.vala:86]: Should really provide a `default:` entry as it cannot be guaranteed that the returned profile be one of those handled.

**Review by jeremypw — COMMENTED** [review]
I like to provide an explicit default entry to open switch clauses, even if it does nothing.  Apart from that, the code looks good.

Needs sign-off from UX team for appearance of checkbox + text + image.

**Review by jeremypw — COMMENTED** [review]
Code looks good - the relevant UX does not appear on my laptop so needs sign-off by another.

**Comment by ryonakano** [comment]
Gtk.RadioButton is removed in GTK4 and it is recommended to use grouped checkbuttons or togglebuttons instead, as per [the migration docs](https://docs.gtk.org/gtk4/migrating-3to4.html#adapt-to-button-class-hierarchy-changes). While GTK4 migration of System Settings and its plugs may not happen soon, I personally felt we might not want to use classes that we know we would need another code and/or UI changes in the near feature. But ignore this comment if this change is intentional.

**Comment by jeremypw** [comment]
I used RadioButtons in a recent PR as it was not obvious how to link generic togglebuttons into a mutually exclusive group in Gtk3.

**Comment by danrabbit** [comment]
Yeah this is a pretty trivial port from RadioButton to CheckButton in Gtk4 :)

**Pull request merged**